public/app/features/dashboard-scene/utils/DashboardModelCompatibilityWrapper.ts

```diff
@@ -1,17 +1,67 @@
-import { DashboardCursorSync, dateTimeFormat, DateTimeInput, EventBusSrv } from '@grafana/data';
-import { behaviors, sceneGraph } from '@grafana/scenes';
+import { Subscription } from 'rxjs';
+
+import { AnnotationQuery, DashboardCursorSync, dateTimeFormat, DateTimeInput, EventBusSrv } from '@grafana/data';
+import { TimeRangeUpdatedEvent } from '@grafana/runtime';
+import { behaviors, SceneDataTransformer, sceneGraph, VizPanel } from '@grafana/scenes';
 
 import { DashboardScene } from '../scene/DashboardScene';
 
+import { findVizPanelByKey, getVizPanelKeyForPanelId } from './utils';
+
 /**
  * Will move this to make it the main way we remain somewhat compatible with getDashboardSrv().getCurrent
  */
 export class DashboardModelCompatibilityWrapper {
-  events = new EventBusSrv();
-  panelInitialized() {}
+  public events = new EventBusSrv();
+  private _subs = new Subscription();
+
+  public constructor(private _scene: DashboardScene) {
+    const timeRange = sceneGraph.getTimeRange(_scene);
+
+    this._subs.add(
+      timeRange.subscribeToState((state, prev) => {
+        if (state.value !== prev.value) {
+          this.events.publish(new TimeRangeUpdatedEvent(state.value));
+        }
+      })
+    );
+  }
+
+  public get id(): number | null {
+    console.error('dashboard.id property is deprecated');
+    return null;
+  }
+
+  public get uid() {
+    return this._scene.state.uid ?? null;
+  }
+
+  public get title() {
+    return this._scene.state.title;
+  }
+
+  public get meta() {
+    return this._scene.state.meta;
+  }
+
+  public get time() {
+    const time = sceneGraph.getTimeRange(this._scene);
+    return {
+      from: time.state.from,
+      to: time.state.to,
+    };
+  }
+
+  /**
+   * Used from from timeseries migration handler to migrate time regions to dashboard annotations
```

Comment:
Do you mean this migration https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/timeseries/migrations.ts#L65 ? Ah it's gonna be interesting to implement as the handler is mutating the original dashboard. Doable of course via setter, but gonna be funky.

Reply:
`__grafanaSceneContext` is a clever idea actually, i like it